### PR TITLE
Move utility functions in common.js into a util object #3978

### LINKED ISSUE #3978: Move utility functions in common.js into a util object
We can do it like below:

(function(window) {
'use strict';
var util = {
add: function(a, b) {
return a + b;
}
}

window.util = util;
})(window));

Reduces global namespace pollution and allows for enabling strict mode.

### PR DESCRIPTION
**Pull request by RatulGhosh**
Fixes #3978

### CODE REVIEW

**Comment by thyageshm** [comment]
@RatulGhosh please try to break the commits into smaller units when there are many changes.

**Comment by RatulGhosh** [comment]
ok sure

File: src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag [hunk 1]
```diff
@@ -23,8 +23,7 @@
    title="<%= Const.Tooltips.FEEDBACK_SESSION_DELETE %>"
    data-toggle="tooltip"
    data-placement="top"
-   <c:if test="${actions.allowedToDelete}">onclick="return toggleDeleteFeedbackSessionConfirmation(${actions.toggleDeleteFeedbackSessionParams});"</c:if>
-   <c:if test="${not actions.allowedToDelete}">onclick="return false"</c:if>
+   onclick="return toggleDeleteFeedbackSessionConfirmation(${actions.toggleDeleteFeedbackSessionParams});"
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: Do you know how changes like this occur? This is definitely not a part of this issue.

**Comment by RatulGhosh** [comment]
@wkurniawan07 I made some changes could you please have a look.

File: src/main/webapp/js/common.js [hunk 2]
```diff
@@ -137,25 +137,14 @@ var TEAMNAME_MAX_LENGTH = 60;
...
-        $tooltips.tooltip('disable');
-    }
-});
+var util = {
 
 /**
  * Checks if the current device is touch based device
  * Reference: https://github.com/Modernizr/Modernizr/blob/master/feature-detects/touchevents.js
  */
-function isTouchDevice() {
```
Comment (bamboo3250) [inline src/main/webapp/js/common.js]: Could you change the indentation to make it follow our Javascript Style Guide? https://docs.google.com/document/d/1gZ6WG6HBTJYHAtVkz9kzi_SUuzfXqzO-SvFnLuag2xM/pub?embedded=true

File: src/main/webapp/js/common.js [hunk 3]
```diff
@@ -137,25 +137,14 @@ var TEAMNAME_MAX_LENGTH = 60;
...
-    return true === (('ontouchstart' in window) || (window.DocumentTouch && document instanceof DocumentTouch));
+isTouchDevice: function() {
+	return true === (('ontouchstart' in window) || (window.DocumentTouch && document instanceof DocumentTouch));
 }
```
Comment (bamboo3250) [inline src/main/webapp/js/common.js]: We should have a comma here. Likewise for all functions below.

File: src/main/webapp/js/common.js [hunk 4]
```diff
@@ -176,18 +165,18 @@ function toggleSort(divElement, comparator) {
     var $selectedDivElement = $(divElement);
     
     if ($selectedDivElement.attr('class') === 'button-sort-none') {
-        sortTable(divElement, colIdx, comparator, true, row);
+        util.sortTable(divElement, colIdx, comparator, true, row);
```
Comment (bamboo3250) [inline src/main/webapp/js/common.js]: Within the object, we can use `this` instead.

**Comment by RatulGhosh** [comment]
My exams were going on so I was unable to make the changes, I will make the changes soon.

File: src/main/webapp/js/instructorFeedbacks.js [hunk 5]
```diff
@@ -14,26 +14,42 @@ function checkFeedbackQuestion(form) {
         if ($(form).find('[name|=' + FEEDBACK_QUESTION_NUMBEROFENTITIESTYPE + ']:checked')
                    .val() === 'custom' &&
                 !$(form).find('.numberOfEntitiesBox').val()) {
+<<<<<<< HEAD
```
Comment (bamboo3250) [inline src/main/webapp/js/instructorFeedbacks.js]: There are a lot of conflicts here. Please resolve them first.

**Comment by thyageshm** [comment]
@RatulGhosh if the PR is ready for review, please post a commit tagging the reviewer

**Comment by RatulGhosh** [comment]
@bamboo3250 I have resolved the conflicts. Could you please review

File: src/main/webapp/js/contextualcomments.js [hunk 6]
```diff
@@ -14,6 +14,9 @@ $(document).ready(function(){
 	
 	function checkComment(form){
 	    var formTextField = $(form).find('[name=commenttext]').val();
+	    if (util.isBlank(formTextField)) {
+	        util.setStatusMessage("Please enter a valid comment. The comment can't be empty.", StatusType.DANGER);
+	        util.scrollToTop();
 	    if (isBlank(formTextField)) {
```
Comment (bamboo3250) [inline src/main/webapp/js/contextualcomments.js]: This part below should be deleted, right?

File: src/main/webapp/js/feedbackSubmissionsEdit.js [hunk 7]
```diff
@@ -757,8 +756,8 @@ function validateAllAnswersHaveRecipient() {
     }
 
     if (!isAllAnswersToMissingRecipientEmpty) {
-        setStatusMessage(statusMessage + '.', StatusType.DANGER);
-    }
+        util.setStatusMessage(statusMessage + '.', StatusType.DANGER);
+	}
```
Comment (bamboo3250) [inline src/main/webapp/js/feedbackSubmissionsEdit.js]: I think you use a tab here. Please change it to 4 spaces.

File: src/main/webapp/js/common.js [hunk 8]
```diff
@@ -90,7 +90,7 @@ var StatusType = {
     WARNING : "warning",
     DANGER : "danger",
     isValidType : function(type) {
-        return type === StatusType.SUCCESS || type === StatusType.INFO || type === StatusType.WARNING || type === StatusType.DANGER;
+	    return type === StatusType.SUCCESS || type === StatusType.INFO || type === StatusType.WARNING || type === StatusType.DANGER;
```
Comment (bamboo3250) [inline src/main/webapp/js/common.js]: 4 spaces should be used here.

File: src/main/webapp/js/common.js [hunk 9]
```diff
@@ -790,15 +790,27 @@ if (!String.prototype.includes) {
...
-function isBlank(str) {
-    if (typeof str !== 'string' && !(str instanceof String)) {
-        return false;
-    }
-    return str.trim() === '';
-}
+/** Selector for status message div tag (to be used in jQuery) */
+var DIV_STATUS_MESSAGE = '#statusMessagesToUser';
+
+
```
Comment (bamboo3250) [inline src/main/webapp/js/common.js]: Hm why there are a lot of empty lines here?

**Comment by RatulGhosh** [comment]
done

**Comment by bamboo3250** [comment]
@RatulGhosh There are a lot of tests failing here. Please run unit tests and investigate them.

image

**Comment by RatulGhosh** [comment]
sure I will look into it

**Comment by wkurniawan07** [comment]
Hi @RatulGhosh, any progress on this PR? It'll be great if you can finish this soon. If you can't finish it by this milestone, let's push it to the next one. However, if you abandon it for too long, I'm afraid I have to close it.

**Comment by RatulGhosh** [comment]
@wkurniawan07 I have synced with the main repo and merged master into the branch but still getting conflicts could you please help.

**Comment by wkurniawan07** [comment]
@RatulGhosh Your `master` branch is behind the main repo's `master` branch, you have to sync it first.

**Comment by wkurniawan07** [comment]
Hi @RatulGhosh, any progress on this PR? I see that the Travis build fails so you can't really push this for review until you fix it. It'll be great if you can finish this soon. If you can't finish it by this milestone, let's push it to the next one. However, if you abandon it for too long, I'm afraid I have to close it.

**Comment by RatulGhosh** [comment]
I was getting some errors while running God mode so couldn't fix the failing tests, I will fix them soon.

**Comment by RatulGhosh** [comment]
Sir, could you please tell where to find resources to learn more about unit tests, I am facing some problem while investigating some of the tests that are failing.

**Comment by wkurniawan07** [comment]
@RatulGhosh what sorts of trouble are you facing? Have you consulted the [troubleshooting guide](https://github.com/TEAMMATES/teammates/blob/master/devdocs/troubleshooting-guide.md) yet?

**Comment by RatulGhosh** [comment]
Sir, I will try my best to finish it within a week, actually caught up with exams.

**Comment by wkurniawan07** [comment]
Closing due to inactivity.

**Pull request closed without merging**